File: Board.app/Contents/Resources/backend/lib/debug.rb
require 'time'

# Log de debug persistant, pour les cas où la console dev (qui se ferme
# avec l'app) ne suffit pas — ex. observer ce que reçoit le backend pendant
# un run de tests, après coup.
#
# Fichier volontairement HORS de ~/Library/Application Support/Board/ (donc
# pas effacé par Tests/version-*/run_tests.sh, qui déplace/restaure ce seul
# dossier autour de chaque run).
#
# Usage : Debug.log("save-app-data reçu, projects-in=#{...}")
module Debug
  LOG_FILE = File.join(Dir.home, 'Library', 'Application Support', 'Board-debug.log')

  module_function

  def log(msg)
    File.open(LOG_FILE, 'a') { |f| f.puts("#{Time.now.strftime('%H:%M:%S.%L')} #{msg}") }
  end
end

File: Sources/Board/Debug.swift
import Foundation

// Même fichier que backend/lib/debug.rb — log de debug partagé, hors du
// dossier de données (jamais déplacé par les tests).
enum Debug {
    static let logFile = NSHomeDirectory() + "/Library/Application Support/Board-debug.log"

    static func log(_ msg: String) {
        let df = DateFormatter()
        df.dateFormat = "HH:mm:ss.SSS"
        let line = "\(df.string(from: Date())) [swift] \(msg)\n"
        guard let data = line.data(using: .utf8) else { return }
        if let handle = FileHandle(forWritingAtPath: logFile) {
            handle.seekToEndOfFile()
            handle.write(data)
            handle.closeFile()
        } else {
            try? data.write(to: URL(fileURLWithPath: logFile))
        }
    }
}
